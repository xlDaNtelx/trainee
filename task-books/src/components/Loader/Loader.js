import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Loader.css';

/**
 * Show loader while books are loading.
 */
class Loader extends Component {
  render() {
    const { loading, opacity } = this.props;
    return (
      <div id="loader" className={loading ? 'show' : 'hide'} style={{ opacity }}>
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
        <div className="dot" />
        <div className="lading" />
      </div>
    );
  }
}

Loader.propTypes = {
  /** State of loading page */
  loading: PropTypes.bool.isRequired,
  /** Opacity for loader container */
  opacity: PropTypes.number.isRequired,
};

export default Loader;
