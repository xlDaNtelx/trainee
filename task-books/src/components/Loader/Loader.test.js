import React from 'react';
import renderer from 'react-test-renderer';
import Loader from './index';

describe('Loader', () => {
  it('renders with loading param', () => {
    const component = renderer.create(
      <Loader loading={Boolean(true)} opacity={0.5} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with opacity', () => {
    const component = renderer.create(
      <Loader opacity={0.5} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
