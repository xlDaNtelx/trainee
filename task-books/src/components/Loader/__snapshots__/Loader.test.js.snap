// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders with loading param 1`] = `
<div
  className="show"
  id="loader"
  style={
    Object {
      "opacity": 0.5,
    }
  }
>
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="lading"
  />
</div>
`;

exports[`Loader renders with opacity 1`] = `
<div
  className="hide"
  id="loader"
  style={
    Object {
      "opacity": 0.5,
    }
  }
>
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="dot"
  />
  <div
    className="lading"
  />
</div>
`;
